main.py

```diff
@@ -200,7 +205,7 @@ def reset_processed_status():
 # ----------------- DISCORD BOT -----------------
 
 
-async def execute_opportunities_webhook(webhook_url, message):
+async def execute_opportunities_webhook(webhook_url, message, internship_message):
```

Comment:
Here is a refactored example of the function that only populates the embed objects if there is a given message.

```py
async def execute_opportunities_webhook(webhook_url, message, internship_message):
    """
    Executes the message which receives the formatted message
    from the format_opportunities() function as well as the webhook
    URL for the respective Discord channel.
    """

    if not message and not internship_message:
        # No messages to send, so skip executing the webhook
        return

    # Create a dictionary payload for the message content
    payload = {
        "tts": False,
        "embeds": [
            {
                "title": f"✧･ﾟ: *✧･ﾟ:* 🎀 {date.today()} 🎀 ✧･ﾟ: *✧･ﾟ:*｡",
                "color": 0xFFFFFF,
            }
        ],
    }

    if message:
        # Add job opportunities embed if a message is provided
        payload["embeds"].append({
            "title": "¸„.-•~¹°”ˆ˜¨ JOB OPPORTUNITIES ¨˜ˆ”°¹~•-.„¸",
            "description": message,
            "color": 0x05A3FF,
        })

    if internship_message:
        # Add internship opportunities embed if a message is provided
        payload["embeds"].append({
            "title": " ¸„.-•~¹°”ˆ˜¨ INTERNSHIP OPPORTUNITIES ¨˜ˆ”°¹~•-.„¸",
            "description": internship_message,
            "color": 0x05A3FF,
        })

    # Convert the payload to JSON format
    json_payload = json.dumps(payload)

    # This will send a POST request to the webhook_url
    response = requests.post(
        webhook_url,
        data=json_payload,
        headers={"Content-Type": "application/json"}
    )

    if response.status_code == 204:
        print("Webhook message was sent successfully!")
    else:
        print(f"Failed to send webhook message. Status Code: {response.status_code}")
```

Reply:
Implemented [here](https://github.com/acmcsufoss/acmOpportunities/pull/11/commits/fdeeb4acca854730a8315bcd1594017d233b4043).